File: php/src/AOC2022/Day3/Day3.php
<?php

namespace AOC2022\Day3;

use AOC\Day;
use AOC\Input;
use Illuminate\Support\Str;

class Day3 extends Day {
    public function part1(Input $input) {
        return $input->lines
            ->flatMap(function($contents) {
                return $contents->splitIn(2)
                    ->map(fn ($compartment) => $compartment->characters()->unique())
                    ->commonElements()
                    ->map(fn ($str) => Str::of($str))
                    ->map(function($item) {
                        return $item->indexInAlphabet() + ($item->isUpper() ? 27 : 1);
                    });
            })
            ->sum();
    }

    public function part2(Input $input) {
        return $input->lines
            ->chunk(3)
            ->flatMap(function($group) {
                return $group->map(function($contents) {
                    return $contents->characters()->unique();
                })
                ->commonElements()
                ->map(fn ($str) => Str::of($str))
                ->map(function($item) {
                    return $item->indexInAlphabet() + ($item->isUpper() ? 27 : 1);
                });
            })
            ->sum();
    }
}

File: swift/Sources/AOC2022/Day3.swift
import AOCKit
import Algorithms

public enum Day3: Solution {
    public static func part1(_ input: String) -> Int {
        input
            .lines
            .flatMap { line in
                return line
                    .evenlyChunked(in: 2)
                    .uniqued()
                    .commonElements()
                    .map { $0.indexInAlphabet + ($0.isUppercase ? 27 : 1) }
            }
            .sum()
    }
    
    public static func part2(_ input: String) -> Int {
        input
            .lines
            .chunks(ofCount: 3)
            .flatMap { chunk in
                chunk
                    .uniqued()
                    .commonElements()
                    .map { $0.indexInAlphabet + ($0.isUppercase ? 27 : 1) }
            }
            .sum()
    }
}
